nautobot/dcim/factory.py

```diff
@@ -0,0 +1,137 @@
+import logging
+import random
+
+import factory
+import factory.fuzzy
+import faker
+
+from nautobot.core.factory import OrganizationalModelFactory, PrimaryModelFactory
+from nautobot.dcim.models import DeviceRole, DeviceType, Manufacturer, Platform
+from nautobot.dcim.choices import SubdeviceRoleChoices
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.factory import random_instance, UniqueFaker
+
+
+logger = logging.getLogger(__name__)
+
+
+# For a randomly deterministic set of vendor names. Must be a tuple.
+MANUFACTURER_NAMES = (
+    "A10",
+    "Arista",
+    "Aruba",
+    "Brocade",
+    "Cisco",
+    "Citrix",
+    "Dell",
+    "F5",
+    "Force10",
+    "Fortinet",
+    "Huawei",
+    "Juniper",
+    "HP",
+    "Palo Alto",
+)
+
+# For `Platform.napalm_driver`, either randomly choose based on Manufacturer slug.
+NAPALM_DRIVERS = {
+    "arista": ["eos"],
+    "cisco": ["ios", "iosxr", "iosxr_netconf", "nxos", "nxos_ssh"],
+    "juniper": ["junos"],
+    "palo-alto": ["panos"],
+}
+
+
+class ManufacturerFactory(OrganizationalModelFactory):
+    class Meta:
+        model = Manufacturer
+        exclude = ("has_description",)
+
+    # Slug isn't defined here since it will always inherit from name.
+    name = UniqueFaker("word", ext_word_list=MANUFACTURER_NAMES)
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("sentence"), "")
```

Comment:
```suggestion
    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
```

Reply:
I had this at first but I have having problems using `description` in filter tests when it was so long, because `text` randomly introduces `\n'`.